instrumentation/couchbase/couchbase-2.0/src/main/java/io/opentelemetry/javaagent/instrumentation/couchbase/v2_0/CouchbaseQueryNormalizer.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.couchbase.v2_0;
+
+import io.opentelemetry.javaagent.instrumentation.api.db.normalizer.ParseException;
+import io.opentelemetry.javaagent.instrumentation.api.db.normalizer.SqlNormalizer;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CouchbaseQueryNormalizer {
+  private static final Logger log = LoggerFactory.getLogger(CouchbaseQueryNormalizer.class);
+
+  private static final Class<?> QUERY_CLASS;
+  private static final Class<?> STATEMENT_CLASS;
+  private static final Class<?> N1QL_QUERY_CLASS;
+  private static final MethodHandle N1QL_GET_STATEMENT;
+  private static final Class<?> ANALYTICS_QUERY_CLASS;
+  private static final MethodHandle ANALYTICS_GET_STATEMENT;
+
+  static {
+    Class<?> queryClass;
+    try {
+      queryClass = Class.forName("com.couchbase.client.java.query.Query");
+    } catch (Exception e) {
+      queryClass = null;
+    }
+    QUERY_CLASS = queryClass;
+
+    Class<?> statementClass;
+    try {
+      statementClass = Class.forName("com.couchbase.client.java.query.Statement");
+    } catch (Exception e) {
+      statementClass = null;
+    }
+    STATEMENT_CLASS = statementClass;
+
+    Class<?> n1qlQueryClass;
+    MethodHandle n1qlGetStatement;
+    try {
+      n1qlQueryClass = Class.forName("com.couchbase.client.java.query.N1qlQuery");
+      n1qlGetStatement =
+          MethodHandles.publicLookup()
+              .findVirtual(
+                  n1qlQueryClass,
+                  "statement",
+                  MethodType.methodType(
+                      Class.forName("com.couchbase.client.java.query.Statement")));
+    } catch (Exception e) {
+      n1qlQueryClass = null;
+      n1qlGetStatement = null;
+    }
+    N1QL_QUERY_CLASS = n1qlQueryClass;
+    N1QL_GET_STATEMENT = n1qlGetStatement;
+
+    Class<?> analyticsQueryClass;
+    MethodHandle analyticsGetStatement;
+    try {
+      analyticsQueryClass = Class.forName("com.couchbase.client.java.analytics.AnalyticsQuery");
+      analyticsGetStatement =
+          MethodHandles.publicLookup()
+              .findVirtual(analyticsQueryClass, "statement", MethodType.methodType(String.class));
+    } catch (Exception e) {
+      analyticsQueryClass = null;
+      analyticsGetStatement = null;
+    }
+    ANALYTICS_QUERY_CLASS = analyticsQueryClass;
+    ANALYTICS_GET_STATEMENT = analyticsGetStatement;
+  }
+
+  public static String normalize(Object query) {
+    if (query instanceof String) {
+      return normalizeString((String) query);
+    }
+    // Couchbase 2.0 uses Query, Couchbase 2.5+ uses Statement
```

Comment:
does it make sense to use muzzle and split out separate instrumentation module for couchbase 2.5?

Reply:
Just a general remark: I think the fewer modules we have, the better :)

Reply:
I think it's a trade-off of how much complexity we need to add in order to handle multiple incompatible versions in a single module

spring-webflux is the only other module that uses MethodHandle, and in that case adds much less complexity compared to this

all other modules currently use muzzle to handle incompatible versions

Reply:
It turns out the situation is much more complicated: I was kinda lazy last time (😅 ) and did not check in which version which class appears, but I did it right now and it looks like we'd need 4 different instrumentations if we wanted to avoid using `MethodHandle`s at all: 2.0.0 - 2.1.0, 2.1.0 - 2.2.0, 2.2.0 - 2.4.3, 2.4.3 - latest.

Reply:
yikes that's a lot of version dependency!

if we keep it in a single module, how do we get test coverage of these different versions? maybe we need a way to specify multiple versions for `testLatestDeps`?

Reply:
Right now all tests use Couchbase 2.5.0 (common test code in `couchbase-testing` for both 2.0.0 and 2.6.0 instrumentations) and every query type except the 2.0.0-2.2.0 `Query` interface is covered. If we wanted to test that we'd probably need to refactor the testing module.

Reply:
thx for explanations, approach / testing coverage seems reasonable to me